alerts/plugins/sso_dashboard.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+#
+# Contributors:
+# Brandon Myers bmyers@mozilla.com
+
+import hjson
+import os
+import sys
+from binascii import b2a_hex
+import boto3
+import datetime
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../../lib'))
+from utilities.logger import logger
+
+
+class message(object):
+    def __init__(self):
+        '''
+        sends certain messages to an SSO dashboard
+        '''
+        # Configuration loading
+        config_file_path = os.path.join(os.path.dirname(__file__), 'sso_dashboard.json')
+        json_obj = {}
+        with open(config_file_path, "r") as fd:
+            try:
+                json_obj = hjson.load(fd)
+            except ValueError:
+                logger.error("FAILED to open the configuration file" + str(config_file_path))
+        self.config = json_obj
+
+        # Set the alert ids based on alert name
+        for name, alert in self.config['alerts'].iteritems():
+            alert['alert_code'] = b2a_hex(name)
+
+        self.connect_db()
+
+        self.registration = '*'
```

Comment:
is this a wildcard registration?

Reply:
@jeffbryner It is, the idea here was that our plugin will potentially want to run on all alerts, and that way, we can be more specific as to what fields we specifically want to match on, and throw to SSO. Ex: we want to match on a category of 'X', but not necessarily if any keys or values have 'X'.

Reply:
Maybe I'm not understanding, but it doesn't seem like it would need to run on all alerts, just geoIP? Unless I'm missing something, it also doesn't match the registration needing to be a list? https://github.com/mozilla/MozDef/blob/ed2f6e02c254a6eb8e60c375a126fb64c27e0a35/alerts/alertWorker.py#L32

Reply:
What I was imaging was in the future, instead of having to modify the actual plugin code, we would only need to modify the configuration file in order to add more alerts that we will be sending. This way, the plugin will be more or less generic, and other folks can possibly use the plugin in their environment.

For the registration, I'm using the new plugin set class from https://github.com/mozilla/MozDef/pull/476  which does support both lists and strings as the value (https://github.com/mozilla/MozDef/pull/476/files#diff-028a2df2c7685ba77c2542250d06e793R72 is a specific test for it, which calls https://github.com/mozilla/MozDef/pull/476/files#diff-90debaea9cca010ebab44a2e8a9652b5 if you're curious).

Reply:
Thanks for the IRL explanation. I missed the part where * is added as a matching value as well. As we discussed, seems like a trade off of a refined registration system vs config file based matching that'll probably play out as we add more types of alerts.